lib/Twig/Extensions/Extension/Gettext.php

```diff
@@ -0,0 +1,184 @@
+        $dnpgettext   = new Twig_Function_Method($this, 'dnpgettext');
+        $dnpfgettext  = new Twig_Function_Method($this, 'dnpfgettext');
+        $dcgettext    = new Twig_Function_Function('dcgettext');
+        $dcfgettext   = new Twig_Function_Method($this, 'dcfgettext');
+        $dcpgettext   = new Twig_Function_Method($this, 'dcpgettext');
+        $dcpfgettext  = new Twig_Function_Method($this, 'dcpfgettext');
+        $dcngettext   = new Twig_Function_Function('dcngettext');
+        $dcnfgettext  = new Twig_Function_Method($this, 'dcnfgettext');
+        $dcnpgettext  = new Twig_Function_Method($this, 'dcnpgettext');
+        $dcnpfgettext = new Twig_Function_Method($this, 'dcnpfgettext');
+        
+        $functions = array(
+            'gettext'      => $gettext,
+            'fgettext'     => $fgettext,
+            'pgettext'     => $pgettext,
+            'pfgettext'    => $pfgettext,
+            'ngettext'     => $ngettext,
+            'nfgettext'    => $nfgettext,
+            'npgettext'    => $npgettext,
+            'npfgettext'   => $npfgettext,
+            'dgettext'     => $dgettext,
+            'dfgettext'    => $dfgettext,
+            'dpgettext'    => $dpgettext,
+            'dpfgettext'   => $dpfgettext,
+            'dngettext'    => $dngettext,
+            'dnfgettext'   => $dnfgettext,
+            'dnpgettext'   => $dnpgettext,
+            'dnpfgettext'  => $dnpfgettext,
+            'dcgettext'    => $dcgettext,
+            'dcfgettext'   => $dcfgettext,
+            'dcpgettext'   => $dcpgettext,
+            'dcpfgettext'  => $dcpfgettext,
+            'dcngettext'   => $dcngettext,
+            'dcnfgettext'  => $dcnfgettext,
+            'dcnpgettext'  => $dcnpgettext,
+            'dcnpfgettext' => $dcnpfgettext
+        );
+        
+        if ($this->useShortnames) {
+            $functions += array(
+                '_'      => $gettext,
+                '_f'     => $fgettext,
+                '_p'     => $pgettext,
+                '_pf'    => $pfgettext,
+                '_n'     => $ngettext,
+                '_nf'    => $nfgettext,
+                '_np'    => $npgettext,
+                '_npf'   => $npfgettext,
+                '_d'     => $dgettext,
+                '_df'    => $dfgettext,
+                '_dp'    => $dpgettext,
+                '_dpf'   => $dpfgettext,
+                '_dn'    => $dngettext,
+                '_dnf'   => $dnfgettext,
+                '_dnp'   => $dnpgettext,
+                '_dnpf'  => $dnpfgettext,
+                '_dc'    => $dcgettext,
+                '_dcf'   => $dcfgettext,
+                '_dcp'   => $dcpgettext,
+                '_dcpf'  => $dcpfgettext,
+                '_dcn'   => $dcngettext,
+                '_dcnf'  => $dcnfgettext,
+                '_dcnp'  => $dcnpgettext,
+                '_dcnpf' => $dcnpfgettext
+            );
+        }
+        
+        return $functions;
+    }
```

Comment:
How is it better to deal with lots of arguments for a function than for a filter ? the issue is the same in both cases.